### PR TITLE
Add Idris2 LSP

### LINKED ISSUE #3934: idris2-lsp
Add idris2-lsp: https://github.com/idris-community/idris2-lsp. Could this interest people?

### PR DESCRIPTION
**Pull request by foxyseta**
## Describe your changes
Add Idris2 LSP.

## Issue ticket number and link
Closes #3934
Old PR: #4009

## Checklist before requesting a review

- [ ] I have successfully tested installation of the package.
- [ ] I have successfully tested the package after installation.

### CODE REVIEW

**Comment by foxyseta** [comment]
(Marked as draft as I haven't been able to properly install the LS in a while now)

**Review by Conarius — APPROVED** [review]

File: packages/idris2-lsp/package.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+homepage: https://github.com/idris-community/idris2-lsp?tab=readme-ov-file
+licenses:
+  - BSD-3-Clause
+languages:
+  - Idris2
+categories:
+  - LSP
+
+source:
+  id: pkg:generic/idris-community/idris2-lsp@nightly-231224
```
[the remark this answers is not in the document]
Comment (foxyseta) [reply packages/idris2-lsp/package.yaml]: Thanks @Conarius! I was using niche package manager `pack` at first, and then forgot to update this. Did you mark your comment as "resolved" by accident? If so, I can easily apply the requested change.

(I will mark this PR as "ready for review" once I actually make this work.)

[the remark this answers is not in the document]
Comment (Conarius) [reply packages/idris2-lsp/package.yaml]: Yeah, it was an accident. But it would be great to make the changes nonetheless.

**Comment by williamboman** [comment]
Hey and thanks for the PR (and sorry for late review)! Custom build scripts are being deprecated because of difficulty in maintaining them over time and their common reliance on system dependencies that Mason cannot detect. You can always host your own registry if you'd like to manage it through Mason, see https://github.com/williamboman/mason-registry-playground/ for a starting point

**Comment by foxyseta** [comment]
Thanks! I released this on [`visimp`'s Mason registry](https://github.com/visimp/mason-registry) as `idris2_ls`.

**Pull request closed without merging**